reactive/swarm.py

```diff
@@ -69,13 +91,108 @@ def swarm_relation_broken():
     c = Compose('files/swarm')
     c.kill()
     c.rm()
-    remove_state('swarm.available')
+    reactive.remove_state('swarm.available')
     status_set('waiting', 'Reconfiguring swarm')
 
-def bind_docker_daemon():
+
+@when('easyrsa installed')
+@when_not('swarm.tls.opensslconfig.modified')
+def inject_swarm_tls_template():
+    """
+    layer-tls installs a default OpenSSL Configuration that is incompatibile
+    with how swarm expects TLS keys to be generated. We will append what
+    we need to the x509-type, and poke layer-tls to regenerate whatever is there.
+    """
+    if not is_leader():
+        return
+    else:
+        status_set('maintenance', 'Reconfiguring SSL PKI configuration')
+
+    log('Updating EasyRSA3 OpenSSL Config')
+    openssl_config = 'easy-rsa/easyrsa3/x509-types/server'
+
+    with open(openssl_config, 'r') as f:
+        existing_template = f.readlines()
+
+    # use list comprehension to enable clients,server usage for certificates
+    # with the docker/swarm daemons.
+    xtype = [w.replace('serverAuth', 'serverAuth, clientAuth') for w in existing_template]  # noqa
+    with open (openssl_config, 'w+') as f:
+        f.writelines(xtype)
+
+    reactive.set_state('swarm.tls.opensslconfig.modified')
+    reactive.set_state('easyrsa configured')
```

Comment:
This set state is unnecessary (just setting modified should prevent this from re-running) and probably not a good idea to set states that the tls layer handles. If another layer wanted to do something on the `easyrsa configured` state it might be more difficult. Minor nit, that does not have to be fixed right now.